packages/lit-dev-discord-bot/src/index.ts

```diff
@@ -0,0 +1,136 @@
+const algClient = algolia(
+  publicVars.algolia.appId,
+  publicVars.algolia.searchOnlyKey
+);
+const index = algClient.initIndex(publicVars.algolia.index);
+
+// The is the GH action secret under the name LIT_DEV_DISCORD_BOT_CLIENT_TOKEN
+const BOT_CLIENT_SECRET = process.env.BOT_CLIENT_SECRET;
+
+if (!BOT_CLIENT_SECRET) {
+  throw new Error('Missing BOT_CLIENT_SECRET');
+}
+
+interface Suggestion {
+  id: number;
+  relativeUrl: string;
+  heading: string;
+  isSubsection: boolean;
+  title: string;
+}
+
+// Build the UI for the slash command.
+const command = new SlashCommandBuilder()
+  .setName('docs')
+  .setDescription('Will search lit.dev.')
+  .addStringOption((option) =>
+    option
+      .setName('query')
+      .setDescription('The query to search for.')
+      .setRequired(true)
+      .setAutocomplete(true)
+  );
+
+const rest = new REST({version: '10'}).setToken(BOT_CLIENT_SECRET);
+
+(async () => {
+  try {
+    console.log('Started refreshing application (/) commands.');
+
+    // Tell Discord that we publish the following slash commands.
+    await rest.put(Routes.applicationCommands(publicVars.discord.clientId), {
+      body: [command],
+    });
+
+    console.log('Successfully reloaded application (/) commands.');
+  } catch (error) {
+    console.error(error);
+  }
+})();
+
+const client = new Client({intents: [GatewayIntentBits.Guilds]});
+
+client.on('ready', () => {
+  console.log(`Logged in as ${client.user?.tag}!`);
+});
+
+client.on('interactionCreate', async (interaction) => {
+  if ((interaction as ChatInputCommandInteraction).commandName !== 'docs') {
+    return;
+  }
+
+  // This happens as the user is typing. Enabled by the SlashCommandBuilder's
+  // .setAutocomplete(true) option.
+  if (interaction.type === InteractionType.ApplicationCommandAutocomplete) {
+    const focusedValue = interaction.options.getFocused();
+    // Do not waste a query if the user has fewer than 3 chars.
+    if (focusedValue.length < 3) {
+      await interaction.respond([]);
+      return;
+    }
+
+    // Search algolia for the query.
+    const searchRes = await index.search<Suggestion>(focusedValue, {
+      page: 0,
+      hitsPerPage: 5,
+    });
+
+    // Transform the hits' relative URL to objects that are readable and
+    // linkable outside of lit.dev.
+    const results = searchRes.hits.map((hit) => {
+      const readableText = hit.isSubsection
+        ? `${hit.title} - ${hit.heading}`
+        : hit.title;
+
+      // autocomplete requires a `name` and a `value` property like a <select>
+      // The name is what is shown to the user, but the `value` is what is
+      // actually sent to the bot in the `.isChatInputCommand()` event.
+      return {
+        name: readableText,
+        value: `https://lit.dev${hit.relativeUrl}`,
+      };
+    });
+
+    await interaction.respond(results);
+  }
+
+  // This is true when the user finally returns a command. (a lit.dev url)
+  if (interaction.isChatInputCommand()) {
```

Comment:
Is this a continuation of the same interaction that was responded to with `await interaction.respond(results)`, or would this be a new interaction?

Reply:
the if statements would be exclusive to each other. I refactored it so PTAL to see if it makes more sense